### PR TITLE
125 auth0

### LINKED ISSUE #125: User authentication

### PR DESCRIPTION
**Pull request by thedavidmeister**
Fixes #125 

Followups:

#137, #138, #139 , #140 , #141 , #142 , #143

### CODE REVIEW

File: test/hl/app/index_test.clj [hunk 1]
```diff
@@ -18,7 +18,7 @@
   [n]
   (element (str "#i" n)))
 
-(defn inputs [] (vec (elements "input")))
+(defn inputs [] (vec (elements "main input")))
```
Comment (thedavidmeister) [inline test/hl/app/index_test.clj]: this is a fix for existing tests targeting `input` elements too broadly

**Comment by thedavidmeister** [comment]
Login:

new-login
popup
login-details-submitted
logged-in
logged-out

**Comment by thedavidmeister** [comment]
error:

screen shot 2016-02-02 at 10 54 21 pm

**Comment by meelijane** [comment]
:shipit:

**Pull request merged**